chain/chunks/src/chunk_cache.rs

```diff
@@ -75,14 +75,17 @@ impl EncodedChunksCache {
     }
 
     // `chunk_header` must be `Some` if the entry is absent, caller must ensure that
-    pub fn get_or_insert_from_header(
+    pub fn get_or_insert_from_header<F>(
         &mut self,
         chunk_hash: ChunkHash,
-        chunk_header: ShardChunkHeader,
-    ) -> &mut EncodedChunksCacheEntry {
+        lazy_chunk_header: F,
```

Comment:
why do we do this instead of passing a reference?

Reply:
Yeah, good point. This is probably just being fancy for no reason. I likely was in the mindset of closures because I was looking at the closure used in the `or_insert_with` call.

Changed to reference in https://github.com/near/nearcore/pull/3715/commits/d74ac85451bbf0d36b9da649b44da104bab80b0b